app/validate_txs.go

```diff
@@ -44,12 +47,22 @@ func FilterTxs(logger log.Logger, ctx sdk.Context, handler sdk.AnteHandler, txCo
 // function used to apply the ante handler.
 func filterStdTxs(logger log.Logger, dec sdk.TxDecoder, ctx sdk.Context, handler sdk.AnteHandler, txs [][]byte) ([][]byte, sdk.Context) {
 	n := 0
+	msgSendTransactionCount := 0
 	for _, tx := range txs {
 		sdkTx, err := dec(tx)
 		if err != nil {
 			logger.Error("decoding already checked transaction", "tx", tmbytes.HexBytes(coretypes.Tx(tx).Hash()), "error", err)
 			continue
 		}
+		msgTypes := msgTypes(sdkTx)
+		if count := countOccurrence(msgTypes, sdk.MsgTypeURL(&types.MsgSend{})); count != 0 {
+			if msgSendTransactionCount+count > v3consts.MsgSendTransactionCap {
+				logger.Debug("skipping tx because the msg send transaction cap was reached", "tx", tmbytes.HexBytes(coretypes.Tx(tx).Hash()))
+				continue
+			}
+			msgSendTransactionCount += count
+		}
```

Comment:
for posterity: iirc we're optimizing for simplicity here, and therefore simply counting the number of messages. in the future, we could use a multidimensional gas price model that would account for the savings of only verifying a single signature for multiple msgs

Reply:
https://github.com/celestiaorg/celestia-app/issues/3948